### PR TITLE
[SPARK-32018][FOLLOWUP][Doc] Add migration guide for decimal value overflow in sum aggregation

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Add migration guide for decimal value overflow behavior in sum aggregation, introduced in https://github.com/apache/spark/pull/29026

### Why are the changes needed?

Add migration guide for the behavior changes from 3.0 to 3.1.
See also: https://github.com/apache/spark/pull/29450#issuecomment-675222779

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Build docs and preview:
image

### CODE REVIEW

File: docs/sql-migration-guide.md [hunk 1]
```diff
@@ -36,6 +36,10 @@ license: |
 
   - In Spark 3.1, NULL elements of structures, arrays and maps are converted to "null" in casting them to strings. In Spark 3.0 or earlier, NULL elements are converted to empty strings. To restore the behavior before Spark 3.1, you can set `spark.sql.legacy.castComplexTypesToString.enabled` to `true`.
 
+  - In Spark 3.1, when `spark.sql.ansi.enabled` is false, sum aggregation of decimal type column always returns `null` on decimal value overflow. In Spark 3.0 or earlier, when `spark.sql.ansi.enabled` is false and decimal value overflow happens in sum aggregation of decimal type column:
+    - If it is hash aggregation with `group by` clause, a runtime exception is thrown.
```
Comment (cloud-fan) [inline docs/sql-migration-guide.md:40]: not many users know the physical nodes. How about
```
In Spark 3.1, Spark always returns null if the sum of decimal overflows under non-ANSI
mode (`spark.sql.ansi.enabled` is false). In Spark 3.0 or earlier, the sum of decimal may
fail at runtime under non-ANSI mode (when the query has GROUP BY and is planned as hash aggregate)
```

Comment (cloud-fan) [inline docs/sql-migration-guide.md:40]: not many users know the physical nodes. How about
```
In Spark 3.1, Spark always returns null if the sum of decimal overflows under non-ANSI
mode (`spark.sql.ansi.enabled` is false). In Spark 3.0 or earlier, the sum of decimal may
fail at runtime under non-ANSI mode (when the query has GROUP BY and is planned as hash aggregate)
```

Comment (gengliangwang) [reply docs/sql-migration-guide.md:40]: The name "non-ANSI mode" is wired. 
Also, we have to mention that Spark 3.0 or earlier also returns null under certain conditions.

Comment (cloud-fan) [reply docs/sql-migration-guide.md:40]: We can use "default mode".

I don't see a difference between "may fail at runtime" or `may return null`. They are mutually exclusive.

Comment (gengliangwang) [reply docs/sql-migration-guide.md]: Thanks, I have updated the doc and screenshot

File: docs/sql-migration-guide.md [hunk 2]
```diff
@@ -36,6 +36,8 @@ license: |
 
   - In Spark 3.1, NULL elements of structures, arrays and maps are converted to "null" in casting them to strings. In Spark 3.0 or earlier, NULL elements are converted to empty strings. To restore the behavior before Spark 3.1, you can set `spark.sql.legacy.castComplexTypesToString.enabled` to `true`.
 
+  - In Spark 3.1, when `spark.sql.ansi.enabled` is false, Spark always returns null if the sum of decimal type column overflows. In Spark 3.0 or earlier, when `spark.sql.ansi.enabled` is false, the sum of decimal type column may return null or incorrect result, or even fails at runtime (depending on the actual query plan execution).
+
```
Comment (maropu) [inline docs/sql-migration-guide.md:40]: nit: we need to describe `spark.sql.ansi.enabled is false` two times? I think its okay just to describe it like this;
```
In Spark 3.0 or earlier, the sum of...
```
or
```
In Spark 3.0 or earlier, in the case, the sum of...
```

**Review by maropu — APPROVED** [review]
LGTM except for the one minor comment.

File: docs/sql-migration-guide.md (hunk 2 above)
Comment (gengliangwang) [reply docs/sql-migration-guide.md:40]: @maropu Thanks

**Comment by gengliangwang** [comment]
Merging to master

**Pull request closed without merging**